app_legacy/Platform/Actions/ResetPlayerAchievementAction.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace LegacyApp\Platform\Actions;
+
+use LegacyApp\Community\Enums\AwardType;
+use LegacyApp\Platform\Enums\AchievementType;
+use LegacyApp\Site\Models\User;
+
+class ResetPlayerAchievementAction
+{
+    public function execute(User &$user, ?int $achievementID = null, ?int $gameID = null): bool
+    {
+        $clause = '';
+        if ($achievementID !== null) {
+            $clause = "AND ach.ID=$achievementID";
+        } elseif ($gameID !== null) {
+            $clause = "AND ach.GameID=$gameID";
+        }
+
+        $query = "SELECT ach.Author, ach.GameID, aw_ach.HardcoreMode,
+                         COUNT(ach.ID) AS Count, SUM(ach.Points) AS Points,
+                         SUM(ach.TrueRatio) AS TruePoints
+                  FROM (
+                     SELECT aw.AchievementID, MAX(aw.HardcoreMode) as HardcoreMode FROM
+                     Awarded aw LEFT JOIN Achievements ach ON ach.ID=aw.AchievementID
+                     WHERE aw.User = :username $clause GROUP BY aw.AchievementID
+                  ) as aw_ach
+                  LEFT JOIN Achievements ach ON ach.ID = aw_ach.AchievementID
+                  WHERE ach.Flags = " . AchievementType::OfficialCore . "
+                  GROUP BY ach.Author, ach.GameID, aw_ach.HardcoreMode";
```

Comment:
This should probably be converted to Eloquent, but I couldn't figure out the correct syntax to make it work the way I needed. Should be easier once we flatten the Awarded table.

Reply:
Makes sense 👍 The transition to Eloquent will require a bit more (performance) testing anyways, as the ORM fetches way too much data by default. So limiting the Eloquent calls to just the fields that we require will be necessary to get the best performance out of it and not decrease it by accident.